File: Programs/Python/ImDaH.py
import sys
sys.path.append("../../Libraries/Python/")
import jpeg

def getFileName():
	done = False
	filename = ""
	while not done:
		filename = str(input("Filename:"))
		try:
			f = open(filename, "rb")
			done = True
			f.close()
		except:
			print("Invalid filename.")
	return filename

programDone = False
while not programDone:
	action = int(input("Quit: 0\nWrite a message: 1\nRead a message: 2\nClear a message: 3\nPNG: 4\nAction:"))
	if action == 0:
		programDone = True
	elif action == 1:
		filename = getFileName()
		jpegWorker = jpeg.JPEG(filename)
		message = str(input("Message:"))
		print(jpegWorker.encode(message))
	elif action == 2:
		filename = getFileName()
		file = open(filename, "rb")
		binaryData = file.read()
		pointerIndex = binaryData.index(bytes.fromhex("FFD9")) + 2
		file.close()
		file = open(filename, "rb")
		file.seek(pointerIndex)
		byteData = file.read()
		message = byteData.decode("UTF-8")
		print(message)
		file.close()
	elif action == 3:
		filename = getFileName()
		file = open(filename, "rb")
		binary = file.read()
		pointerIndex = binary.index(bytes.fromhex("FFD9")) + 2
		file.close()
		file = open(filename, "rb")
		done = False
		binaryData = b""
		while not done:
			byte = file.read(1)
			binaryData += byte
			pos = file.tell()
			if pos == pointerIndex:
				done = True
				break
		file.close()
		file = open(filename, "wb")
		file.write(binaryData)
		file.close()
	elif action == 4:
		filename = getFileName()
		file = open(filename, "rb")
		binaryData = file.read()
		pointerIndex = binaryData.index(bytes.fromhex("49454E44AE426082")) + 8
		file.seek(pointerIndex)
		byteData = file.read()
		message = byteData.decode("UTF-8")
		print(message)
		file.close()

	print("\n")

File: Programs/C/ImDaH.c
#include <stdio.h>
#include <string.h>
#include "../../Libraries/C/lib.h"

void clearSTDIN() {
	char buf[1001];
	fgets(buf, 1001, stdin);
}

char *getInput(bufferLength) {
	char NEWLINE[1] = "\n";
	char buf[bufferLength];
	fgets(buf, bufferLength, stdin);
	printf("Raw is '%s'\n", buf);
	char cleaned[bufferLength];
	int cleanedIndex = 0;
	for (int i = 0; i < bufferLength; i++) {
		if (buf[i] == NEWLINE[0]) {
			printf("[NEWLINE]");
		} else {
			printf("%c", buf[i]);
			cleaned[cleanedIndex] = buf[i];
			cleanedIndex += 1;
		}
	}
	printf("\nCleaned is %s\n", cleaned);
	for (int i = 0; i < bufferLength; i++) {
		printf("[%c]", cleaned[i]);
	}
	clearSTDIN();
	return(cleaned);
}

int main() {
	int done = 0;
	while (done == 0) {
		system("pwd");
		printf("\nActions:\n(0): Write a message.\n(1): Read a message.\n(2): Clear messages.\n(3): Quit.\n:");
		char action;
		char actions[5] = "01234";
		scanf(" %c", &action);
		getchar();
		printf("%c\n", action);
		if (action == actions[0]) {
			encode();
		} else if (action == actions[1]) {
			decode();
		} else if (action == actions[2]) {
			strip();
		} else if (action == actions[3]) {
			done = 1;
		} else if (action == actions[4]) {
			system("ls");
			done = 1;
		}
	}
	return(0);
}